### PR TITLE
Add validation for lifecycle prefix tags conflict

### PR DESCRIPTION
**Pull request by bhavyaKhatri2703**
Addresses #12187.

This PR adds validation for lifecycle prefix tags and provides an option to remove one of the tags.
Example:
building:destroyed,  building

image

### CODE REVIEW

File: modules/osm/tags.js [hunk 1]
```diff
@@ -433,3 +431,15 @@ export const osmUrlKeys = new Set([
     'website:menu',
     'post_office:website',
 ]);
+
+export const osmLifecycleConflictPrefixes = new Set([
```
Comment (RudyTheDev) [inline modules/osm/tags.js:435]: What is this list based on? The prefixes documented on https://wiki.openstreetmap.org/wiki/Lifecycle_prefix are slightly different.

Why isn't this just using `osmLifecyclePrefixes`, why a separate list?

**Comment by RudyTheDev** [comment]
What about something like `disused:leisure=stadium` + `leisure=park` - these are not mutually exclusive.

Or what about something like `disused:amenity=parking` + `disused:parking=surface` - this would retain `disused:parking` value.

This should really have unit tests for common scenarios/examples and edge cases.

**Comment by tordans** [comment]
The next, planned step for lifecycle tags is https://github.com/openstreetmap/iD/pull/10844
IMO any improvements should be based on the progress over there as an improvement to what is already planned.
